plugins/mcreator-core/blockly/mcreator_extensions.js

```diff
@@ -203,3 +198,129 @@ Blockly.Extensions.register('dimension_custom_list_provider',
         this.appendDummyInput().appendField(new Blockly.FieldDropdown(
             arrayToBlocklyDropDownArray(javabridge.getListOf("dimension_custom"))), 'dimension');
     });
+

```

Comment:
I think this should be in its own file, like block_mcitem.js and field_ai_condition.js are to keep things nice and tidy